packages/trpc/server/routers/viewer/bookings/get.schema.ts

```diff
@@ -2,7 +2,45 @@ import { z } from "zod";
 
 import { ZTextFilterValue } from "@calcom/features/data-table/lib/types";
 
-export const ZGetInputSchema = z.object({
+// Define type first to use with z.ZodType annotation
+// This prevents full Zod generic tree from being emitted in .d.ts files
+// Note: offset has .default(0), so input type has it optional but output type has it required
+type BookingStatus = "upcoming" | "recurring" | "past" | "cancelled" | "unconfirmed";
+
+type TGetInputSchemaFilters = {
+  teamIds?: number[];
+  userIds?: number[];
+  status?: BookingStatus;
+  statuses?: BookingStatus[];
+  eventTypeIds?: number[];
+  attendeeEmail?: string | z.infer<typeof ZTextFilterValue>;
+  attendeeName?: string | z.infer<typeof ZTextFilterValue>;
+  bookingUid?: string;
+  afterStartDate?: string;
+  beforeEndDate?: string;
+  afterUpdatedDate?: string;
+  beforeUpdatedDate?: string;
+  afterCreatedDate?: string;
+  beforeCreatedDate?: string;
+};
+
+// Input type - what callers send (offset is optional)
+export type TGetInputSchemaInput = {
```

Comment:
nit: Maybe we call it TGetInputRawSchema to make it a bit clearer in name.  
InputSchemaInput is a bit weird